### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by jedgarpark**

### CODE REVIEW

File: Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    (12, 0, 127),
+    (13, 0, 127),
+    (14, 0, 127),
+    (15, 0, 127)
+]
+
+# Create the input objects list for potentiometers
+#  by getting pin # attributes using string formatting
+for k in range(knob_count):
+    knb[k] = (AnalogIn(getattr(board, "A{}".format(knb[k][0]))), knb[k][1], knb[k][2])
```
Comment (ladyada) [inline Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py]: dont re-assign knb[] - make a new array for the settings thats not shared

File: Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        offset = 0
+    idx = int(map_range((ctl + offset) & 0xFF00, 1200, 65500, 0, ctrl_max))
+    if idx != old_idx:  # if index changed, adjust hysteresis offset and set flag
+        # offset is 25% of the control slice (65536/ctrl_max)
+        offset = int(0.25 * sign(idx - old_idx) * (65535 / ctrl_max))  # edit 0.25 to adjust slices
+    return idx, offset
+
+def sign(x):  # determine the sign of x
+    if x >= 0:
+        s = 1
```
Comment (ladyada) [inline Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py]: return 1 here

File: Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    else:
+        s = -1
```
Comment (ladyada) [inline Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py]: ditto

File: Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    return s
+
+while True:
+    # read all the knob values
+    for i in range(knob_count):
+        cc_val[i] = ctl_idx(knb[i][0].value, knb[i][2] - knb[i][1] + 1, cc_val[i][0], cc_val[i][1])
```
Comment (ladyada) [inline Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py]: make this code easier to read, there's no need to use short variables, name things so others can follow!

File: Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py (hunk 2 above)
Comment (jedgarpark) [reply Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py]: will do

File: Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py (hunk 3 above)
Comment (jedgarpark) [reply Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py]: will do

File: Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py (hunk 1 above)
Comment (jedgarpark) [reply Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py]: OK, I'm on it.

File: Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py (hunk 4 above)
Comment (jedgarpark) [reply Grand_Central_MIDI_Knobs/Grand_Central_MIDI_Knobs.py]: right on!

**Comment by jedgarpark** [comment]
OK, ready for review.

**Pull request merged**